### PR TITLE
Create OCR API

### PR DESCRIPTION
**Pull request by zdeveloper**
# Pull Request

## Description

- Add OCR fastAPI
- Add tests
- fix broken tests

note the docker integration will come in a followup PR

## Related Issues
#180

## Additional Notes
[Add any additional context or notes that reviewers should know about.]

## Checklist
Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.
- [ ] I have notified teammates in the review thread to build awareness.

## Checklist for Reviewers
Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: OCR/ocr/api.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from ocr.services.image_segmenter import ImageSegmenter, segment_by_color_bounding_box
+
+app = FastAPI()
+segmenter = ImageSegmenter(
+    segmentation_function=segment_by_color_bounding_box,
+)
+ocr = ImageOCR()
+
+
+@app.get("/")
```
Comment (arinkulshi-skylight) [inline OCR/ocr/api.py:17]: You can probably remove this.

File: OCR/ocr/api.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+async def hello_world():
```
Comment (arinkulshi-skylight) [inline OCR/ocr/api.py:18]: you can probably remove this.

Comment (zdeveloper) [reply OCR/ocr/api.py:18]: I actually added it for health checks

Comment (zdeveloper) [reply OCR/ocr/api.py:18]: I am happy to rename it

Comment (zdeveloper) [reply OCR/ocr/api.py:18]: I can rename it to make its purpose more obvious

File: OCR/ocr/services/image_segmenter.py [hunk 3]
```diff
@@ -79,33 +76,42 @@ def segment_by_color_bounding_box(self) -> dict[str, np.ndarray]:
...
 
-        self.labels = labels
+        labels = json.load(open(labels_path, "r"))
 
         if self.debug is True:
             self.debug_folder = "debug_segments"
             os.makedirs(self.debug_folder, exist_ok=True)
-            print(f"raw_image shape: {self.raw_image.shape}")
-            print(f"segmentation_template shape: {self.segmentation_template.shape}")
+            print(f"raw_image shape: {raw_image.shape}")
```
Comment (arinkulshi-skylight) [inline OCR/ocr/services/image_segmenter.py:114]: In a following tickets we should change these to logging
logging.debug(f"raw_image shape: {self.raw_image.shape}")

File: OCR/ocr/api.py (hunk 2 above)
Comment (zdeveloper) [reply OCR/ocr/api.py]: Done, thanks!

**Review by arinkulshi-skylight — APPROVED** [review]
LGTM
